test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SchemaCompare/SchemaCompareServiceTests.cs

```diff
@@ -495,5 +496,140 @@ private void ValidateSchemaCompareScriptGenerationWithExcludeIncludeResults(Sche
             string afterIncludeScript = generateScriptOperation.ScriptGenerationResult.Script;
             Assert.True(initialScript.Length == afterIncludeScript.Length, $"Changes should be same as inital since we included what we excluded, before {initialScript}, now {afterIncludeScript}");
         }
+
+        /// <summary>
+        /// Verify opening an scmp comparing two databases
+        /// </summary>
+        [Fact]
+        public async void SchemaCompareOpenScmpDatabaseToDatabaseRequest()
+        {
+            SqlTestDb sourceDb = await SqlTestDb.CreateNewAsync(TestServerType.OnPrem, false, null, SourceScript, "SchemaCompareOpenScmpSource");
+            SqlTestDb targetDb = await SqlTestDb.CreateNewAsync(TestServerType.OnPrem, false, null, TargetScript, "SchemaCompareOpenScmpTarget");
+
+            try
+            {
+                // create schema compare scmp file that should be opened
+                string filePath;
+                SchemaCompareDatabaseEndpoint sourceEndpoint = new SchemaCompareDatabaseEndpoint(sourceDb.ConnectionString);
+                SchemaCompareDatabaseEndpoint targetEndpoint = new SchemaCompareDatabaseEndpoint(targetDb.ConnectionString);
+                SchemaCompareOpenScmpResult result = this.CreateAndOpenScmp(sourceEndpoint, targetEndpoint, out filePath);
+
+                Assert.Equal(targetDb.DatabaseName, result.OriginalTargetName);
+                Assert.Equal(sourceDb.DatabaseName, result.SourceEndpointInfo.DatabaseName);
+                Assert.Equal(targetDb.DatabaseName, result.TargetEndpointInfo.DatabaseName);
+                Assert.Equal(SchemaCompareEndpointType.Database, result.SourceEndpointInfo.EndpointType);
+                Assert.Equal(SchemaCompareEndpointType.Database, result.TargetEndpointInfo.EndpointType);
+                Assert.Empty(result.ExcludedSourceElements);
+                Assert.Empty(result.ExcludedTargetElements);
```

Comment:
Nit : can we combine some of the code to common functions in the three tests - they seems to be almost very similar other than dacpac/db distinction...

Reply:
Sure, made this change in the latest changes